src/test/java/com/sap/cloudfoundry/client/facade/CloudControllerClientIntegrationTest.java

```diff
@@ -0,0 +1,125 @@
+package com.sap.cloudfoundry.client.facade;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.apache.commons.codec.binary.StringUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+
+import com.sap.cloudfoundry.client.facade.domain.CloudSpace;
+
+class CloudControllerClientIntegrationTest {
+
+    private static CloudControllerClient clientWihtoutTarget;
+    private static CloudControllerClient client;
+
+    @BeforeAll
+    static void login() throws MalformedURLException {
```

Comment:
I think it will be very nice, if there is a class with common utility methods e.i. `login`, `printVariable`, `printHeader`, etc.

Reply:
I would leave it at the time being in this way. When refactoring is needed and more classes are needed, e.g. one class for apps, one class for services, etc. then it could be extracted.